### PR TITLE
Adding remove_obsidian_comments postprocessor to remove obsidian format comments

### PR DESCRIPTION
**Pull request by Thompson-Jason**
Adding feature requested in issue #158 to remove Obsidian comments. As mentioned in PR #159 by @zoni I don't think it would be difficult to expand this postprocessor to instead of removing the comments making them HTML comments.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -60,6 +60,12 @@ struct Opts {
         default = "false"
     )]
     hard_linebreaks: bool,
+    #[options(
+        no_short,
+        help = "Remove Obsidian style comments from exported file",
+        default = "false"
+    )]
+    remove_obsidian_comments: bool,
```
Comment (zoni) [inline src/main.rs:68]: I think I would like the CLI interface for this to be `--comments=[keep-unchanged|remove]`.

This would allow it to be expanded to  `--comments=[keep-unchanged|remove|as-html-comments]` in the future (and "as-html-comments" should probably become the default at that point).

File: src/postprocessors.rs [hunk 2]
```diff
@@ -51,6 +52,70 @@ fn filter_by_tags_(
...
+
+                output.pop();
+
+                if inside_comment {
+                    inside_comment = false;
+                    continue;
+                }
+
+                if !text.eq(&CowStr::from("%%")) {
+                    let re = Regex::new(r"%%.*?%%").unwrap();
```
Comment (zoni) [inline src/postprocessors.rs:87]: Can this be implemented without having to use a regex?

If it does need to be done via a regex, the construction should be wrapped in something like a [LazyCell](https://doc.rust-lang.org/std/cell/struct.LazyCell.html) to only compile it once instead of for each iteration of the loop, as regex compilation is computationally quite heavy.

File: tests/testdata/input/remove-comments/test_comments.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+%% Comment should be removed %%
+```md
+%% Comment in code block should be kept %%
+```
+
+%%
+This is
+a block comment
+that should be removed
+%%
```
Comment (zoni) [inline tests/testdata/input/remove-comments/test_comments.md:11]: Does your current implementation handle non-text elements within the comment? As in, Obsidian treats this entire block as a comment:

````
%%

None of the below should be visible in output:

## Header

Code block:

```
foo
```

---

%%
````

**Review by zoni — COMMENTED** [review]
Thanks for picking this up! Here's some non-exhaustive, preliminary comments from a quick review. :smile: 

Let me know if any of this is unclear and needs more explanation

File: src/main.rs (hunk 1 above)
Comment (Thompson-Jason) [reply src/main.rs:68]: I like this implementation suggested and will get this changed over. However, I disagree when it comes to changing the default in the future. I think leaving the comments unchanged has been the way the tool works since the beginning and people would be expecting the tool to run this way potentially breaking workflows. I think it would be best practice to makes additions like this opt-in rather than opt-out.

File: tests/testdata/input/remove-comments/test_comments.md (hunk 3 above)
Comment (Thompson-Jason) [reply tests/testdata/input/remove-comments/test_comments.md:11]: While my current implementation handles most of this edge case it struggles with the code block (removing everything except the opening and closing ```. I will get started on this fix.

**Comment by Thompson-Jason** [comment]
@zoni  was there something more you felt this needed?
